enhancements/support-less-on-more.md

```diff
@@ -0,0 +1,146 @@
+upstream components tracked in that release do a bugfix release as
+well.
+
+The idea of the proposal is to make it more appealing to get the
+latest release. The only challenge can be the
+`MIN_KUBERNETES_VERSION`, as discussed below.
+
+### Support down to `MIN_KUBERNETES_VERSION`
+
+Each version we release is supported on **any** OpenShift version that
+satisfy the following:
+
+- OpenShift version is still supported.
+- The pipeline `MIN_KUBERNETES_VERSION` is compatible with the
+  kubernetes version of OpenShift.
+
+In a gist this would mean, today, that:
+
+- 1.11 (`MIN_KUBERNETES_VERSION` = 1.24) is supported on >= 4.11
+- 1.10 (`MIN_KUBERNETES_VERSION` = 1.23) is supported on >= 4.10
+
+As of today, no OpenShift version below 4.10 is supported, so we could
+support only 1.10 and 1.11. In the case of an OpenShift version is
+still supported but the `MIN_KUBERNETES_VERSION` is lower than any of
+the 2 most recent OpenShift Pipeline's release, we would have to
+support that one (in maintenance/security mode) until the given
+OpenShift release is not supported anymore, but only on that given
+OpenShift version.
+
+For example, when we release 1.12 (`MIN_KUBERNETES_VERSION` = 1.24),
+we will need to still support 1.10 *on* 4.10 (not necessarily on 4.11
+or higher).
+
+## Testing
+
+This means we are testing OpenShift Pipelines on more OpenShift version. Although this increase a little bit the testing effort for one release, the assumption here are the following:
+
+- We'll have more and more automated test making that effort less.
+- We'll have less z-stream for older version to do, and thus reducing the overall effort.
+
+## Dependencies
+
+There is possible dependencies from/to our operator that could be impacted by this move.
+One of them is the *devconsole*, aka the UI of openshift-pipelines. Supporting older version of OpenShift might mean incompatibilities with the UI and bugs that would require an z-stream, and that might be a problem. One assumption here is that **we would move forward to own our UI (using dynamic plugin) so that we are independent of the console/devconsole lifecycle**.
+
+## Examples
+
+- OSP 1.11 - released 2023-06
+	- OCP >= 4.11
+	- Not supported in 2023-12 (OSP 1.13 + 1 month)
+- OSP 1.12 - released 2023-09
+	- OCP >= 4.11
+	- Not supported in 2024-04 (OSP 1.14 + 1 month)
+- OSP 1.13 - release 2023-11
+	- OCP >= 4.12 (EUS)
+	- Not supported in 2024-06 (OSP-1.15 + 1 month) **or** until 4.12 EUS is not supported (which is, for 4.12, **2025-01**)
+- OSP 1.14 - release 2024-03 (OSP 1.16 + 1 month)
+	- (*example*) OCP >= 4.13
+	- Not supported in 2024-09 (OSP 1.16 + 1 month)
+- OSP 1.15 - release 2024-06 (OSP 1.17 + 1 month)
+	- (*example*) OCP >= 4.13
+	- Not supported in 2024-12 (OSP 1.18 + 1 month)
+- … and so one
+
+In this example, OSP 1.13 would need to be supported, *in maintenance mode*, until 17th January 2025 as it's the "highest" version that works on 4.12, which is an EUS.
```

Comment:
i see this as us having to support 3 versions of pipelines still due to EUS constraints, even if it will be maintenance support 🤔

Reply:
Yes, there will be no way around this. It depends on how quick or slow tekton moves the minimum kubernetes version value though.

Reply:
OCP 4.12
- maintenance support ends: July 17, 2024
- EUS support ends: January 17, 2025

we are releasing 1.16 on September 17, 2024, which is "after" maintenance support but "within" EUS support window AND min kube version is satisfied
- we will not release 1.16 on OCP 4.12
- we will continue supporting the latest available release (1.15) on OCP 4.12 till EUS support ends

i.e. for a given EUS release, we support the latest shipped pipelines version that was shipped before maintenance support ended and we will not ship a new one

OR

we are releasing 1.16 on September 17, 2024, which is "after" maintenance support but "within" EUS support window AND min kube version is satisfied
- we will release 1.16 on OCP 4.12
IF ELSE min kube version is not satisfied, then continue support 1.15 on OCP 4.12

---

OCP 4.13
- maintenance support ends: November 17, 2024	

we are releasing 1.17 on December 31, 2024, then:
- we will ship it for OCP 4.12 but we will not ship it for OCP 4.13
- we will ship it for OCP 4.14 onwards (whatever the supported OCP versions are)